pkg/apis/serving/v1alpha1/inference_graph.go

```diff
@@ -335,3 +337,25 @@ type InferenceGraphList struct {
 func init() {
 	SchemeBuilder.Register(&InferenceGraph{}, &InferenceGraphList{})
 }
+
+func (ig *InferenceGraph) HasAuthEnabled() bool {
+	return ig.GetAnnotations()[constants.EnableAuthAnnotationKey] == "true"
+}
+
+// ResolveServiceAccountName returns the ServiceAccount name to use for the InferenceGraph pod.
+// If the user has customized the name of the ServiceAccount, the custom name is used. If there is no
+// custom ServiceAccount the returned name depends on whether the IG is auth-protected:
+//   - If the IG is *not* auth-protected, `default` is returned (backwards compatibility)
+//   - If the IG is *not* auth-protected, the returned value is the name of the IG with `-auth-verifier` suffix
```

Comment:
Maybe one is protected and the other is not?

Reply:
Yes, the second bullet is for auth-protected ones. I'll fix it and remove the `*not*`